Patch release 1.25.06

d-torrance opened the pull request:
*Target date*: June 1, 2025

- [x] Fix for `random(Module, Module)` bug
- [x] #3822
- [x] #3823
- [x] #3831
- [x] #3840 (to enable building Fedora 42 package)
- [x] Fix for #3851
- [ ] Updated citation info?
- [ ] Anything else?

mahrud committed:
fixed Leah Gold's name in Macaulay2Doc contributors

bkholler committed:
added updates from main MultigradedImplicitization repo including interpolation and faster multidegree computation

bkholler committed:
removed debugging mode

mahrud committed:
removed Citation for packages with no authors

mahrud committed:
added release date on package documentation page

mahrud committed:
fixed minor issues in NoetherNormalization package

mahrud committed:
combine different Sasha Zotines

mahrud committed:
combine two Timothy Duffs

mahrud committed:
add M2 citation guide to Macaulay2Doc

bkholler committed:
fixed tests in trimBasisInDegree

bkholler committed:
updated test for interpolateComponent

mahrud committed:
fixed a bug in random

d-torrance committed:
Bump version number to 1.25.06

d-torrance committed:
Run builds on PR's to *stable* (no longer master)

d-torrance committed:
Update branch name (master -> stable) in package citations

d-torrance committed:
Update branch name (master -> stable) in package documentation

[ci skip]

mahrud commented:
Definitely #3823.

mahrud committed:
added slightly better random(Module,Module) strategy

mahrud commented:
To be clear, we don't want to include new packages, etc. correct? How about package patches? e.g. #3822.

d-torrance commented:
I would think no new packages or features, but I'm no too picky.  If you think it would be good to have, then I'm down!

mahrud committed:
updated version of Eigen in CMake to 3.4.0

d-torrance committed:
Add patch for building lrslib in GCC 15

[ci skip]

mahrud commented:
By the way, one of the steps in the release should be updating https://github.com/Macaulay2/language-macaulay2, because GitHub syntax highlighting uses it.

d-torrance commented:
Oh cool -- I always wondered how GitHub did syntax highlighting!

mahrud committed:
removed Verbose from exports of MultigradedImplicitization

d-torrance committed:
Have packageFiles return empty list for packages w/o files

In particular, packages defined in startup.m2 (currently just User),
stdio, or currentString.

This was also causing a problem with "check User", which was running
into this issue via "needsPackage", so we add a unit test

d-torrance committed:
Add warning message when reloading package

Each type from the package is redefined, so existing instances won't
work any more.

d-torrance committed:
Fix realpath/toAbsolutePath for stdio/currentString

Not actual files, so don't prepend a directory
